data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -59,43 +58,57 @@ public SimpleDataService(
         this.dataReader = dataReader;
     }
 
-    public CompletableFuture<Optional<DataReaderRequest>> authoriseRequest(final DataRequest dataRequest) {
+    /**
+     * Query for the references.  It will return the information needed to retrieve the resources.  If there is no
+     * data to be returned, a {@link ForbiddenException} is thrown.
+     *
+     * @param dataRequest data provided by the client for requesting the resource
+     * @return reference to the resources that are to be returned to client
+     */
+    public CompletableFuture<AuthorisedData> authoriseRequest(final DataRequest dataRequest) {
         LOGGER.debug("Querying persistence for token {} and resource {}", dataRequest.getToken(), dataRequest.getLeafResourceId());
-        CompletableFuture<Optional<AuthorisedRequestEntity>> futureRequestEntity = this.persistenceLayer.getAsync(dataRequest.getToken(), dataRequest.getLeafResourceId());
+        CompletableFuture<Optional<AuthorisedRequestEntity>> futureRequestEntity = persistenceLayer.getAsync(dataRequest.getToken(), dataRequest.getLeafResourceId());
         return futureRequestEntity.thenApply(maybeEntity -> maybeEntity.map(
-                entity -> new DataReaderRequest()
-                        .context(entity.getContext())
-                        .user(entity.getUser())
-                        .resource(entity.getLeafResource())
-                        .rules(entity.getRules())
-                )
+                entity -> AuthorisedData.Builder.create()
+                        .withResource(entity.getLeafResource())
+                        .withUser(entity.getUser())
+                        .withContext(entity.getContext())
+                        .withRules(entity.getRules())
+                ).orElseThrow(() -> new ForbiddenException("There is no data for this request."))
         );
     }
 
-    public Pair<AtomicLong, AtomicLong> read(final DataReaderRequest readerRequest, final OutputStream out) {
-        final AtomicLong recordsProcessed = new AtomicLong(0);
-        final AtomicLong recordsReturned = new AtomicLong(0);
-
-        LOGGER.debug("Reading from reader with request {}", readerRequest);
-        DataReaderResponse readerResponse = this.dataReader.read(readerRequest, recordsProcessed, recordsReturned);
+    /**
+     * Includes the resources into the OutputStream that is to be provided to the client
+     *
+     * @param authorisedData the information for the resources in the context of the request
+     * @param out                an {@link OutputStream} to write the stream of resources to (after applying rules)
+     * @param recordsProcessed   number of records that have been processed
+     * @param recordsReturned    number of records that have been returned
+     * @return boolean indicating that the process has been successful
```

Comment:
Does true/false indicate success?
```suggestion
     * @return true if the process has been successful
```

Reply:
Done, easier in the code along with other changes.